mbuild/packing.py

```diff
@@ -57,12 +61,26 @@ def fill_box(compound, n_compounds, box, overlap=0.2, seed=12345):
                          "packmol.exe is on the path.")
         raise IOError(msg)
 
-    box = _validate_box(box)
+    if box is not None:
+        box = _validate_box(box)
     if not isinstance(compound, (list, set)):
         compound = [compound]
-    if not isinstance(n_compounds, (list, set)):
+    if n_compounds is not None and not isinstance(n_compounds, (list, set)):
         n_compounds = [n_compounds]
 
+    if density is not None:
```

Comment:
We should probably have some descriptive error messages if the correct arguments are not provided.  For example, if the user only provides `compound` and nothing else, they should receive an error message that some combination of two of `n_compounds`, `box`, and `density` must be provided.  Likewise an error should be raised if `n_compounds`, `box`, and `density` are all provided.